app/controllers/provider_interface/provider_agreements_controller.rb

```diff
@@ -0,0 +1,25 @@
+module ProviderInterface
+  class ProviderAgreementsController < ProviderInterfaceController
+    before_action :set_page_name
+
+    def new_data_sharing_agreement
```

Comment:
This controller sounds very RESTy,
we might just want to use `new, create, show` actions,
what do you think?

Reply:
It's a nod to RESTful conventions, but the model and the controller are meant to be general purpose (ProviderAgreement instead of DataSharingAgreement), while the routes are very specific (`/provider/data-sharing-agreement/new`). So I've made the controller action names specific, too.

Reply:
What other agreements are we going to have?

Reply:
We only have data sharing agreements at the mo but the intention is to replace Docusign when onboarding SCITTs.